api/src/main/java/com/redhat/ipaas/api/v1/rest/DataManager.java

```diff
@@ -101,20 +103,27 @@ public void init() {
         }
     }
 
-    public void addToCache(ModelData modelData) throws ClassNotFoundException, IOException {
-        Class<? extends WithId> clazz = getClass(modelData.getModel());
-        Map<String, WithId> entityMap = cache.computeIfAbsent(modelData.getModel().toLowerCase(), k -> new HashMap<>());
-        logger.debug(modelData.getModel() + ":" + modelData.getData());
-        WithId entity = clazz.cast(mapper.readValue(modelData.getData(), clazz));
-        Optional<String> id = entity.getId();
-        String idVal;
-        if (!id.isPresent()) {
-            idVal = generatePK(entityMap);
-            entity = entity.withId(idVal);
-        } else {
-            idVal = id.get();
+    public void addToCache(ModelData modelData) {
+        try {
+            Class<? extends WithId> clazz;
+            clazz = getClass(modelData.getModel());
+            Map<String, WithId> entityMap = cache.computeIfAbsent(modelData.getModel().toLowerCase(), k -> new HashMap<>());
+            logger.debug(modelData.getModel() + ":" + modelData.getData());
+            WithId entity = clazz.cast(mapper.readValue(modelData.getData(), clazz));
+            Optional<String> id = entity.getId();
+            String idVal;
+            if (!id.isPresent()) {
+                idVal = generatePK(entityMap);
+                entity = entity.withId(idVal);
+            } else {
+                idVal = id.get();
+            }
+            entityMap.put(idVal, entity);
+        } catch (ClassNotFoundException e) {
```

Comment:
Since we don't handle them differently we could possibly collapse them into Exception. No biggie though!